ui_testing/testcases/basic_tests/test06_orders.py

```diff
@@ -130,52 +130,23 @@ def test003_cancel_button_edit_departments(self, save):
-            The user cannot archive an order unless all corresponding analysis are archived
-            LIMS-3425
-
-            New: Archive order has active analysis
-            The user cannot archive an order unless all corresponding analysis are archived
-            LIMS-4329
-        :return:
-        """
-        order_row = self.order_page.get_random_order_row()
-        self.order_page.click_check_box(source=order_row)
-        order_data = self.base_selenium.get_row_cells_dict_related_to_header(
-            row=order_row)
-        analysis_numbers_list = order_data['Analysis No.'].split(',')
-        self.base_selenium.LOGGER.info(
-            ' + Try to archive order with number : {}'.format(order_data['Order No.']))
-        order_deleted = self.order_page.archive_selected_orders(
-            check_pop_up=True)
-        self.base_selenium.LOGGER.info(' + {} '.format(order_deleted))
-
-        if order_deleted:
-            self.base_selenium.LOGGER.info(
-                ' + Order number : {} deleted successfully'.format(order_data['Order No.']))
-            self.analyses_page.get_analyses_page()
-            self.base_selenium.LOGGER.info(
-                ' + Assert analysis numbers : {} is not active'.format(analysis_numbers_list))
-            has_active_analysis = self.analyses_page.search_if_analysis_exist(
-                analysis_numbers_list)
-            self.base_selenium.LOGGER.info(
-                ' + Has activated analysis? : {}.'.format(has_active_analysis))
-            self.assertFalse(has_active_analysis)
-        else:
-            self.analyses_page.get_analyses_page()
-            self.base_selenium.LOGGER.info(
-                ' + Archive Analysis with numbers : {}'.format(analysis_numbers_list))
-            self.analyses_page.search_by_number_and_archive(
-                analysis_numbers_list)
-            self.order_page.get_orders_page()
-            rows = self.order_page.search(analysis_numbers_list[0])
-            self.order_page.click_check_box(source=rows[0])
-            self.base_selenium.LOGGER.info(
-                ' + archive order has analysis number =  {}'.format(analysis_numbers_list[0]))
-            self.order_page.archive_selected_orders()
-            rows = self.order_page.result_table()
-            self.assertEqual(len(rows), 1)
+        '''
+        LIMS-6516
+        User can archive a main order
+        '''
+        # Get random order
+        order_request = self.orders_api.get_all_orders(limit=20).json()
+        self.assertEqual(order_request['status'], 1)
+        orders_records = order_request['orders']
+        self.assertNotEqual(len(orders_records), 0)
+        random_order_index = self.generate_random_number(lower=0, upper=len(orders_records) - 1)
+        order_data = orders_records[random_order_index]
+        order_id = order_data['id']
+
+
+        # Arcihve main order
+        archived_order = self.orders_api.archive_main_order(order_id)
```

Comment:
in this test case steps you need to archive main order using UI not API

Reply:
Done